src/build_queue.rs

```diff
@@ -274,79 +301,70 @@ impl BuildQueue {
     pub fn get_new_crates(&self, index: &Index) -> Result<usize> {
         let mut conn = self.db.get()?;
         let diff = index.diff()?;
-        let (mut changes, oid) = diff.peek_changes()?;
+        let (mut changes, oid) = diff.peek_changes_ordered()?;
         let mut crates_added = 0;
 
         // I believe this will fix ordering of queue if we get more than one crate from changes
         changes.reverse();
 
         for change in &changes {
-            match change {
-                Change::Yanked(release) => {
-                    let res = conn
-                        .execute(
-                            "
-                            UPDATE releases
-                                SET yanked = TRUE
-                            FROM crates
-                            WHERE crates.id = releases.crate_id
-                                AND name = $1
-                                AND version = $2
-                            ",
-                            &[&release.name, &release.version],
-                        )
-                        .with_context(|| {
-                            format!(
-                                "error while setting {}-{} to yanked",
-                                release.name, release.version
-                            )
-                        });
-                    match res {
-                        Ok(_) => debug!("{}-{} yanked", release.name, release.version),
-                        Err(err) => report_error(&err),
-                    }
-                    if let Err(err) = cdn::invalidate_crate(&self.config, &self.cdn, &release.name)
-                    {
-                        report_error(&err);
-                    }
+            if let Some((ref krate, ..)) = change.deleted() {
+                match delete_crate(&mut conn, &self.storage, &self.config, krate)
+                    .with_context(|| format!("failed to delete crate {}", krate))
+                {
+                    Ok(_) => info!(
+                        "crate {} was deleted from the index and will be deleted from the database",
```

Comment:
"will be deleted" isn't quite right, is it? this is *after* we've already deleted it in the database.